src/features/views/components/MoveViewDialog.tsx

```diff
@@ -0,0 +1,266 @@
+      {(viewTree) => {
+        const folders = getAllParentFolderIds(viewedFolder, viewTree)
+          .map((id) => folderById(id, viewTree))
+          .filter((folder): folder is ZetkinViewFolder => !!folder);
+
+        // Add the root folder to the beginning
+        const breadcrumbItems = [
+          {
+            id: null,
+            title: messages.browserLayout.title(),
+          },
+          ...folders,
+        ];
+        return (
+          <Breadcrumbs
+            aria-label="breadcrumb"
+            itemsAfterCollapse={2}
+            itemsBeforeCollapse={1}
+            maxItems={3}
+            separator={<NavigateNextIcon fontSize="small" />}
+          >
+            {breadcrumbItems.map(({ title, id }) =>
+              id == viewedFolder ? (
+                <Box key={id ?? 'root'}>{title}</Box>
+              ) : (
+                <Link
+                  key={id ?? 'root'}
+                  color="inherit"
+                  onClick={() => onClickFolder(id)}
+                  sx={{ cursor: 'pointer' }}
+                  underline="hover"
+                >
+                  {title}
+                </Link>
+              )
+            )}
+          </Breadcrumbs>
+        );
+      }}
+    </ZUIFuture>
+  );
+};
+
+export type MoveViewDialogProps = {
+  close: () => void;
+  itemToMove: ViewBrowserItem;
+};
+const MoveViewDialog: FunctionComponent<MoveViewDialogProps> = ({
+  close,
+  itemToMove,
+}) => {
+  const [viewedFolder, setViewedFolder] = useState(itemToMove.folderId);
+
+  const { orgId } = useNumericRouteParams();
+  const itemsFuture = useViewBrowserItems(orgId, viewedFolder);
+  const viewTreeFuture = useViewTree(orgId);
+  const { moveItem } = useViewBrowserMutations(orgId);
+
+  const messages = useMessages(messageIds);
+  const theme = useTheme();
+  const fullScreen = useMediaQuery(theme.breakpoints.down('md'));
+
+  if (itemToMove.type == 'back') {
+    throw new Error('Should not be possible to move a back button');
+  }
+
+  const doMove = (folderId: number | null) => {
+    moveItem(itemToMove.type, itemToMove.data.id, folderId);
+    close();
+  };
+
+  return (
+    <Dialog
+      fullScreen={fullScreen}
+      fullWidth
+      maxWidth={'sm'}
+      onClose={close}
+      open={true}
+      PaperProps={{
+        sx: {
+          height: '80vh',
+          maxHeight: '600px',
+        },
+      }}
+      scroll="paper"
+    >
+      <DialogTitle>
+        <MoveItemBreadcrumbs
+          onClickFolder={setViewedFolder}
+          orgId={orgId}
+          viewedFolder={viewedFolder}
+        />
+      </DialogTitle>
+
+      <DialogContent>
+        <ZUIFuture future={itemsFuture}>
+          {(data) => {
+            const relevantItems = data.filter(
+              // This explicit typing should not be necessary, but it is. Don't know why.
+              (item): item is ViewBrowserFolderItem | ViewBrowserViewItem =>
+                item.type != 'back'
```

Comment:
I believe this may have changed in some later version of TS, but in the version we're using you have to filter "by type predicate" (like you're doing here) for TS to understand that the resulting Array will have a narrower type than the one that `filter()` is invoked on.

Reply:
Gotcha! Should I remove the comment if this is a common occurrence in the project?

Reply:
I've found a more explicit and self-documenting solution, and removed the comment while I was at it. PTAL!